pandas/tests/frame/methods/test_append.py

```diff
@@ -165,23 +165,23 @@ def test_append_dtypes(self):
         df2 = DataFrame({"bar": np.nan}, index=range(1, 2))
         result = df1.append(df2)
         expected = DataFrame(
-            {"bar": Series([Timestamp("20130101"), np.nan], dtype="M8[ns]")}
+            {"bar": Series([Timestamp("20130101"), np.nan], dtype="object")}
```

Comment:
Changing to ignore the all-NaN columns (although following the casting rules for common type of Timestamp and float) is still a breaking change, I would say. Certainly given that np.nan is generally used as missing value indicator (and allowed in other situations for datetime dtype).